extensions/vertx-common/runtime/src/main/java/io/quarkus/vertx/common/runtime/VertxCommonProducer.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.vertx.common.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.EventBus;
+
+/**
+ * Produces a configured Vert.x instance.
+ * It also exposes the Vert.x event bus.
+ */
+@ApplicationScoped
+public class VertxCommonProducer {
+
+    private volatile Vertx vertx;
+
+    void initialize(Vertx vertx) {
+        this.vertx = vertx;
+    }
+
+    @Singleton
+    @Produces
+    public Vertx vertx() {
+        return vertx;
+    }
+
+    @Singleton
+    @Produces
+    public EventBus eventbus() {
```

Comment:
Should be moved to user-facing. 
Core extension should not use the event bus and if they do, they would retrieve it using `vertx.eventBus()`

Reply:
Done.